### PR TITLE
Require python >= 3.7

### PR DESCRIPTION
**Pull request by maxnoe**
This adresses #1466 , so dropping 3.6 for the next release.

There was also a bug in `ci/install.sh` resulting in not installing the correct python version for conda builds (so we didn't even test on 3.6 since conda switched to 3.7 by default)

### CODE REVIEW

**Review by kosack — DISMISSED** [review]

**Comment by dneise** [comment]
There is a timeout in `conda env create -n cta-dev --file environment.yml` on Travis when building for Py3.8 

Is this reason for concern? I restarted the build this morning once .. hoping it was just a mishap on travis.... no luck

File: ci/install.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ if [[ "$CONDA" == "true" ]]; then
     # Useful for debugging any issues with conda
     conda info -a
 
-	sed -i -e "s/- python=.*/- python=$TRAVIS_PYTHON_VERSION/g" environment.yml
+	sed -i -e "s/- python=.*/- python=$PYTHON_VERSION/g" environment.yml
```
Comment (dneise) [inline ci/install.sh:26]: I am confused by this change ... looks independent of the Py3.6 -> Py 3.7 step .. was this a bug before?

Comment (maxnoe) [reply ci/install.sh:26]: Yes it was. It was revealed in #1472 because it should have error on 3.6 but didn't since it really was using 3.7

**Comment by maxnoe** [comment]
@dneise it worked yesterday with the first commit... Ah, but that probably was still using the wrong version.

**Comment by maxnoe** [comment]
It is a missing eventio for python 3.8 I think.

**Comment by dneise** [comment]
Erm .. I see to be missing the point here ... it is still failing no?

**Comment by maxnoe** [comment]
Yes, because there is no conda package of eventio for python 3.8. @kosack will upload it shortly.

**Comment by dneise** [comment]
Ah I see ... thanks

File: .travis.yml [hunk 2]
```diff
@@ -54,6 +44,16 @@ matrix:
...
+#    - os: linux
+#      dist: bionic
+#      language: python
+#      python: 3.9
+#      addons:
+#        apt:
+#          packages:
+#            - pandoc
+#            - graphviz
+
```
Comment (vuillaut) [inline .travis.yml:56]: why leaving this commented?

Comment (maxnoe) [reply .travis.yml:56]: I'll remove it. python 3.9 does not work yet

File: ci/install.sh [hunk 3]
```diff
@@ -23,9 +23,9 @@ if [[ "$CONDA" == "true" ]]; then
     # Useful for debugging any issues with conda
     conda info -a
 
-	sed -i -e "s/- python=.*/- python=$TRAVIS_PYTHON_VERSION/g" environment.yml
-	travis_wait 20 conda env create -n cta-dev --file environment.yml
-	conda activate cta-dev
+	sed -i -e "s/- python=.*/- python=$PYTHON_VERSION/g" environment.yml
+	travis_wait 20 conda env create -n travis cta-dev --file environment.yml
```
Comment (vuillaut) [inline ci/install.sh:27]: remove `cta-dev`

**Review by vuillaut — CHANGES_REQUESTED** [review]

File: ci/install.sh (hunk 3 above)
Comment (maxnoe) [reply ci/install.sh:27]: how did it work???

**Review by vuillaut — APPROVED** [review]

**Review by dneise — APPROVED** [review]

**Pull request merged**